third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -100,13 +103,13 @@ func resourceDataflowJob() *schema.Resource {
 				ValidateFunc: validation.StringInSlice([]string{"cancel", "drain"}, false),
```

Comment:
Hi @c2thorn, does this mean that changes to `on_delete` would also trigger an update-by-replacement?

Reply:
Hi, yes that is true. Previously any changes to `on_delete` would destroy/recreate all jobs, but now it will trigger the update-by-replacement for streaming jobs, while retaining the same destroy/recreate functionality for batch.

Reply:
Oh that's interesting that we used to ForceNew on that! It probably should be something that can be updatable completely on its own, without triggering a ForceNew or an update-by-replacement.

Reply:
Ah yeah I was surprised to see it used to be `ForceNew` as well.

I agree that this field should be updatable on its own without triggering a ForceNew or update-by-replacement. Would you like me to file this as a separate issue?

Reply:
Created https://github.com/terraform-providers/terraform-provider-google/issues/6301 to track

Reply:
Thanks!